src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java

```diff
@@ -0,0 +1,1899 @@
+     * @param startPos Genomic start position (1-based, inclusive) of the variant.
+     * @param refEnd Genomic end position (1-based, inclusive) of the reference allele.
+     * @param altEnd Genomic end position (1-based, inclusive) of the alternate allele.
+     * @return {@code true} if replacing the reference with the alternate results in a frameshift.  {@code false} otherwise.
+     */
+    public static boolean isFrameshift(final int startPos, final int refEnd, final int altEnd) {
+
+        final int refLength = refEnd - startPos + 1;
+        final int altLength = altEnd - startPos + 1;
+
+        // We know it's a frameshift if we have a replacement that is not of a
+        // length evenly divisible by 3 because that's how many bases are read at once:
+        return ((Math.abs( refLength - altLength ) % 3) != 0);
+    }
+
+    /**
+     * Gets the position describing where the given allele and variant lie inside the given transcript using transcript-based coordinates.
+     * The index will be calculated even if the given variant ends outside the bounds of the given transcript.
+     * Assumes {@code transcript} is a sorted list (in exon number order).
+     * @param variant A {@link Locatable} to locate inside the given {@code transcript}.
+     * @param transcript A sorted {@link List} of {@link Locatable} (in exon number order) that describe the transcript to use for locating the given {@code allele}.
+     * @param strand The strand on which the transcript is read.
+     * @return The position (1-based, inclusive) describing where the given {@code allele} lies in the given {@code transcript}.  If the variant is not in the given {@code transcript}, then this returns -1.
+     */
+    public static int getStartPositionInTranscript( final Locatable variant,
+                                                    final List<? extends Locatable> transcript,
+                                                    final Strand strand) {
+        Utils.nonNull(variant);
+        Utils.nonNull(transcript);
+        assertValidStrand( strand );
+
+        int position = 1;
+
+        boolean foundPosition = false;
+
+        final SimpleInterval variantStartLocus;
+        if ( strand == Strand.POSITIVE ) {
+            variantStartLocus = new SimpleInterval(variant.getContig(), variant.getStart(), variant.getStart());
+        }
+        else {
+            variantStartLocus = new SimpleInterval(variant.getContig(), variant.getEnd(), variant.getEnd());
+        }
+
+        for (final Locatable exon : transcript) {
+            if (!exon.getContig().equals(variantStartLocus.getContig())) {
+                throw new GATKException("Variant and transcript contigs are not equal: "
+                        + variantStartLocus.getContig() + " != " + exon.getContig());
+            }
+
+            if (new SimpleInterval(exon).contains(variantStartLocus)) {
```

Comment:
Since I'm dealing with both `forward` and `backward` strands, I'm creating the new `SimpleInterval` so I can reference that object directly (as opposed to adding a few more if statements).

I'll add a comment.